#151511475 Users should be authenticated via the user interface

gbengaoyetade opened the pull request:
#### What does this PR do?
Authenticates users from the front end

#### Description of Task to be completed?
Users, once they fill the signup or login in form correctly, they should be authenticated and redirected to the gaming screen.

#### How should this be manually tested?
Visit the app and try signing up or loging in

#### What are the relevant pivotal tracker stories?
Users should be authenticated via the user interface 

#### Screenshots (if appropriate)
![slice](https://user-images.githubusercontent.com/8515311/31056376-828faf0e-a6c8-11e7-887c-a0203f5d6561.png)

#### Questions:
None

gbengaoyetade committed:
Merge branch 'develop' of https://github.com/andela/rainier-cfh into develop

gbengaoyetade committed:
Merge branch 'develop' of https://github.com/andela/rainier-cfh into develop

gbengaoyetade committed:
feature(user-authentication): add user login and begin signup

gbengaoyetade committed:
feature(user-authentication): finish up frontend sign up
This commit also modifies some links that were not working as a result of merging pull requests
Finishes #151511475

gbengaoyetade committed:
Merge branch 'develop' into feature/151511475/frontend-user-authentication

segunolalive reviewed (commented):
Good job 👍 

Ensure a user cannot return to the signin/signup page after logging in

gbengaoyetade committed:
refactor(authentication): add route protection
This commit makes signup and signin unaccessible when user is authenticated. It also make dashboard unaccessible when user is not authenticated

gbengaoyetade committed:
refactor(authentication): add logout functionality

ibonly commented on public/js/controllers/index.js:
```diff
@@ -21,4 +21,51 @@ angular.module('mean.system')
+        $scope.error = error.error;
+       });
+      };
+      $scope.validateInput = (userInput) => {
```
Empty line between 2 function

ibonly commented on public/js/controllers/index.js:
```diff
@@ -21,4 +21,51 @@ angular.module('mean.system')
+        }
+        return true;
+      }
+      $scope.signup = (userInput) => {
```
Empty line between 2 function

gbengaoyetade replied:
Done

ibonly reviewed (changes requested):
Your PR lack test, update the test.

gbengaoyetade committed:
refactor(authentication): update lougout functionality
Logout now redirects to singin page instead of the landing page

gbengaoyetade committed:
Merge branch 'feature/151511475/frontend-user-authentication' of https://github.com/andela/rainier-cfh into feature/151511475/frontend-user-authentication

gbengaoyetade commented:
Tests written by @enodi covers this PR
